test/test_data_reader.rb

```diff
@@ -137,6 +139,31 @@ def test_input_file_as_stdin
     end
   end
 
+  def test_errors
```

Comment:
`test_errors` seems large and that tells me that it's testing too many different scenarios and that it could probably broken down into multiple test cases